tsmo/src/modified_spat_plotter.py

```diff
@@ -0,0 +1,97 @@
+            modified_spat_data['Epoch_Time(ms)'] = (modified_spat_data['Moy'] * 60000) + modified_spat_data['Timestamp'] + first_day_epoch #convert moy to milliseconds              
+            modified_spat_data['Epoch_Time(s)'] = modified_spat_data['Epoch_Time(ms)'] / 1000
+            min_time = dt.datetime.fromtimestamp(modified_spat_data['Epoch_Time(s)'].min())
+            max_time = dt.datetime.fromtimestamp(modified_spat_data['Epoch_Time(s)'].max())
+
+            fig, ax1 = plt.subplots()
+            fig.set_size_inches(10, 10)
+
+            #Iterate through every signal group in the parsed file
+            for group in signal_groups:
+                df = modified_spat_data.copy()
+                group_subset = df[df['Signal_Group'] == group]
+                dates=[dt.datetime.fromtimestamp(ts) for ts in group_subset["Epoch_Time(s)"]]               
+
+                #iterate through group data and plot based on current and next state
+                for i in range(0, len(group_subset)-1):
+                    #red state
+                    if (group_subset['Event_State'].iloc[i] == 3)&(group_subset['Event_State'].iloc[i+1] == 3):
+                        time1 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i])
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='red', linewidth=10)
+                    #green state
+                    elif (group_subset['Event_State'].iloc[i] == 6)&(group_subset['Event_State'].iloc[i+1] == 6):
+                        time1 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i])
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='green', linewidth=10)
+                    #yellow state
+                    elif (group_subset['Event_State'].iloc[i] == 8)&(group_subset['Event_State'].iloc[i+1] == 8):
+                        time1 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i])
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='yellow', linewidth=10)
+                    #change in state from red to green, draw green
+                    elif (group_subset['Event_State'].iloc[i] == 3)&(group_subset['Event_State'].iloc[i+1] == 6):
+                        time1 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i])
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='green', linewidth=10)
+
+                        #add vertical line for green state change
+                        ax1.vlines(time2, 0, 9, linestyles='dashed', color='green', linewidth=1)
```

Comment:
I thought we removed these green vertical lines. Where are you place these. Just at green starts/end. If so you can probably remove this vertical line.

Reply:
I left them for now, but yea I think the plan for the future is to have the ET zones displayed for each vehicle.